vectorizers/transformers.py

```diff
@@ -502,6 +527,126 @@ def fit_transform(self, X, y=None, **fit_params):
 
         for sequence in X:
             seq = np.array(sequence)
-            result.append(seq[self.offset:] - seq[: -self.offset])
+            result.append(seq[self.offset :] - seq[: -self.offset])
 
         return result
+
+
+class CategoricalColumnTransformer(BaseEstimator, TransformerMixin):
+    """
+    This transformer is useful for describing an object as a bag of the categorical values that
+    have been used to represent it within a pandas DataFrame.
+
+    It takes an categorical column name to groupby, object_column_name, and one or more categorical columns to be used
+    to describe these objects, descriptor_column_name.  Then it returns a Series with an index being the unique entries
+    of your object_column_name and the values being a list of the appropriate categorical values from your
+    descriptor_column_name.
+
+    It can be thought of as a PivotTableTransformer if you'd like.
+    Parameters
+    ----------
+    object_column_name: string
+        The column name from the DataFrame where our object values can be found.
+        This will be the thing we are grouping by.
+
+    descriptor_column_name: string or list
+        The name or names of the categorical column(s) who's values will be used for describing our
+        objects.  If you are using multiple names it's recommended that you set include_column_name=True.
+
+    include_column_name: bool (default = False)
+        Should the column name be appended at the beginning of each value?
+        This is useful if you intend to combine values from multiple categorical columns
+        after the fact.
+
+    unique_values: bool (default = False)
+        Should we apply a unique to the values in column before building our list representation?
+
+    """
+
+    def __init__(
+        self,
+        object_column_name,
+        descriptor_column_name,
+        include_column_name=False,
+        unique_values=False,
+    ):
+        self.object_column_name = object_column_name
+        self.descriptor_column_name = descriptor_column_name
+        # Get everything on consistent footing so we don't have to handle multiple cases.
+        if type(self.descriptor_column_name) == str:
+            self.descriptor_column_name_ = list(self.descriptor_column_name)
```

Comment:
Calling list on a string will turn it into a list of the single characters that make up the string. I'm not sure if this is what you want. Probably you mean
```python
self.descriptor_column_name_ = [self.descriptor_column_name]
```

Reply:
I do indeed.  I just hit this error when writing the documentation this weekend.  You are correct sir.